Celeste.Mod.mm/Mod/UI/TextMenuExt.cs

```diff
@@ -355,5 +355,839 @@ public class IntSlider : TextMenu.Item {
+                Container.Add(item.SelectWiggler = Wiggler.Create(0.25f, 3f, null, false, false));
+                item.ValueWiggler.UseRawDeltaTime = (item.SelectWiggler.UseRawDeltaTime = true);
+                if (Selection == -1) {
+                    FirstSelection();
+                }
+                RecalculateSize();
+                item.Added();
+                return this;
+            }
+            /// <summary>
+            /// Remove any non-submenu <see cref="TextMenu.Item"/> from the Submenu
+            /// </summary>
+            /// <param name="item">Item to be removed</param>
+            /// <returns></returns>v
+            public SubMenu Remove(TextMenu.Item item) {
+                if (Container != null) {
+                    orig_Remove(item);
+                    return this;
+                } else {
+                    delayedAddItems.Remove(item);
+                    return this;
+                }
+            }
+            private SubMenu orig_Remove(TextMenu.Item item) {
+                int num = Items.IndexOf(item);
+                if (num == -1) {
+                    return this;
+                }
+                Items.RemoveAt(num);
+                item.Container = null;
+                Container.Remove(item.ValueWiggler);
+                Container.Remove(item.SelectWiggler);
+                RecalculateSize();
+                return this;
+            }
+            public void Clear() {
+                Items = new List<TextMenu.Item>();
+            }
+            public int IndexOf(TextMenu.Item item) {
+                return Items.IndexOf(item);
+            }
+            public void FirstSelection() {
+                Selection = -1;
+                MoveSelection(1, false);
+            }
+            public void MoveSelection(int direction, bool wiggle = false) {
+                int selection = Selection;
+                direction = Math.Sign(direction);
+                int num = 0;
+                using (List<TextMenu.Item>.Enumerator enumerator = Items.GetEnumerator()) {
+                    while (enumerator.MoveNext()) {
+                        if (enumerator.Current.Hoverable) {
+                            num++;
+                        }
+                    }
+                }
+                bool flag = num > 2;
+                do {
+                    Selection += direction;
+                    if (flag) {
+                        if (Selection < 0) {
+                            Selection = Items.Count - 1;
+                        } else if (Selection >= Items.Count) {
+                            Selection = 0;
+                        }
+                    } else if (Selection < 0 || Selection > Items.Count - 1) {
+                        Selection = Calc.Clamp(Selection, 0, Items.Count - 1);
+                        break;
+                    }
+                }
+                while (!Current.Hoverable);
+                if (!Current.Hoverable) {
```

Comment:
What even is whitespace? In other news, this looks mildly confusing at first glance.